paasta_tools/tron_tools.py

```diff
@@ -125,6 +125,17 @@ def get_value(self, key, args, kwds):
                 return Formatter.get_value(key, args, kwds)
 
 
+def get_tron_api_endpoints(cluster: str):
+    endpoints = load_system_paasta_config().config_dict.get('tron_api_endpoints', {})
+    if cluster not in endpoints.keys():
+        raise Exception(f"tron api endpoint is not defined for cluster {cluster}")
+    return endpoints[cluster]
+
+
+def get_tron_client_for_cluster(cluster: str):
+    return TronClient(get_tron_api_endpoints(cluster))
```

Comment:
the way this function is used, you're going to create a client `num_clusters * num_services * num_instances` times. since client is only different per-cluster - you're doing the same work a lot. you could do memoization of `get_tron_client_for_cluster` but that's not great for long-lived processes (when configs change some of client objects may become stale). i would suggest to create these client objects much earlier in the call stack and pass as argument to functions that need them, such as `tron_instance_status`.

this will be tricky as the best candidate, `report_status_for_cluster`, is already invoked `num_clusters * num_services` times, but it is a cli function, so you don't have to worry about cache invalidation there - you can make a global dict `TRON_CLIENTS[cluster]` and only create a client object for a given cluster if it's not yet in the dict.